packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -65,15 +65,22 @@ export const useField = <
 
   const focusTimeoutRef = React.useRef<NodeJS.Timeout | undefined>(undefined);
 
-  const handleInputClick = useEventCallback((...args) => {
-    onClick?.(...(args as []));
-
+  const syncSelectionWithDOM = () => {
```

Comment:
Just a small refacto to avoid calling the `onClick` prop during focus.
Otherwise it receives a `FocusEvent` which is incorrect.